### PR TITLE
Add support for Trivy as an SBOM analyzer

### PR DESCRIPTION
**Pull request by sambetts-cisco**
Add support for Trivy as an SBOM analyzer

This the first part of an effort to introduce Trivy support to
kubeclarity.

This commit adds Trivy as an SBOM analyzer which can be used in
conjunction with the other SBOM analyzers or by itself to scan the
contents of container images or directories and produce a merged result.

Along with Trivy we introduce a new input type "rootfs" which allows
kubeclarity to scan operating system root file systems like unpacked
containers or virtual machine volumes.

### CODE REVIEW

File: shared/pkg/analyzer/trivy/trivy.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			},
+			ScanOptions: trivyFlag.ScanOptions{
+				Target:         userInput,
+				SecurityChecks: nil, // Disable all security checks for SBOM only scan
+			},
+			ReportOptions: trivyFlag.ReportOptions{
+				Format:       "cyclonedx", // Cyconedx format for SBOM so that we don't need to convert
+				ReportFormat: "all",       // Full report not just summary
+				Output:       &output,     // Save the output to our local buffer instead of Stdout
+				ListAllPkgs:  true,        // List all packages for full SBOM
```
Comment (akpsgit) [inline shared/pkg/analyzer/trivy/trivy.go:76]: Needed to look what this means for trivy. maybe better to state "List all packages regardless of vulnerabilities"

File: shared/pkg/analyzer/trivy/trivy.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			return
+		}
+
+		err := artifact.Run(context.TODO(), trivyOptions, trivySourceType)
+		if err != nil {
+			a.setError(res, fmt.Errorf("failed to generate SBOM: %w", err))
+			return
+		}
+
+		frm := formatter.New(formatter.CycloneDXJSONFormat, output.Bytes())
```
Comment (akpsgit) [inline shared/pkg/analyzer/trivy/trivy.go:102]: shouldn't this be CycloneDX XML? i see that the merge sbom logic doing formatting, but maybe better to have a uniform return format?

**Comment by sambetts-cisco** [comment]
Moved back to draft after identifying issues with standardising the cyclonedx from trivy, hashes etc are missing from the component.

File: shared/pkg/analyzer/trivy/trivy.go (hunk 1 above)
Comment (sambetts-cisco) [reply shared/pkg/analyzer/trivy/trivy.go]: updated the comment

**Review by akpsgit — DISMISSED** [review]

**Review by pbalogh-sa — DISMISSED** [review]

**Review by pbalogh-sa — APPROVED** [review]

**Pull request merged**